### DISCUSSION TITLE
example of uploading uncompressed local file to s3 as gzip

### DISCUSSION THREAD

**Discussion by tooptoop4** [RaRe-Technologies/smart_open#745] [General]
I have local uncompressed file, I would like to save its gzip version on s3. but don't want to store the gzip locally

**Comment by mpenkov** [comment]
Basic example:

```python
with open('local_file', 'rb') as fin:
   with open('s3://bucket/key.gz', 'wb') as fout:
       fout.write(fin.read())
```

**Reply by piskvorky** [reply]
…and to make use of streaming, for large inputs:

```python
with open('local_file', 'rb') as fin:
   with open('s3://bucket/key.gz', 'wb') as fout:
       for line in fin:
           fout.write(line)
```

smart_open was made for easier streaming of primarily text files, with text transformations and other actions in between. So just-copy-the-file-verbatim isn't its primary use case. I expect the native `aws cli` + `gzip` piping will do a better job there.

**Reply by tooptoop4** [reply]
do u have example with sed/bash transform on each row?
